codalab/lib/download_manager.py

```diff
@@ -181,12 +201,17 @@ def summarize_file(self, uuid, path, num_head_lines, num_tail_lines, max_line_le
         """
         if self._is_available_locally(uuid):
             file_path = self._get_target_path(uuid, path)
-            string = file_util.summarize_file(file_path, num_head_lines, num_tail_lines, max_line_length, truncation_text)
+
+            # If the bundle is not running, the contents are immutable and we can save time by going to the cache
+            key = (file_path, num_head_lines, num_tail_lines, max_line_length, truncation_text)
+            summarize = lambda: file_util.summarize_file(*key)
+            string = self._cached_if_stable(uuid, 'summarize_file', key, summarize)
```

Comment:
Unlike `target_info`, we might want to change the implementation of `summarize_file`...do you imagine just being able to blow away the cache at any time and be fine?